openlibrary/plugins/upstream/utils.py

```diff
@@ -734,6 +734,25 @@ def canonical_url(self):
         """
         readable_path = web.ctx.get('readable_path', web.ctx.path) or ''
         query = web.ctx.query or ''
+
+        if len(query) > 0:
+            queries_to_exclude = ['sort', 'mode', 'v', 'type', 'debug']
+            parameters = query.split('&')
+            pattern = re.compile('.*?(?==)')
+            params_to_remove = []
+
+            for param in parameters:
+                p = pattern.match(param).group().replace('?', '')
+                if p in queries_to_exclude:
+                    params_to_remove.append(param)
+
+            for p in params_to_remove:
+                parameters.remove(p)
+
+            query = '&'.join(parameters)
+            if '?' not in query and len(query) > 0:
+                query = '?' + query
+
         host = web.ctx.host or ''
         url = (host + readable_path + query)
         return ("https://" + url) if url else ''
```

Comment:
The module has an `unparse` method too. https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlunparse